lib/model.d.ts

```diff
@@ -2497,9 +2497,9 @@ export abstract class Model {
    * @param options.reset Clear all previously set data values
    */
   set<K extends keyof this>(key: K, value: this[K], options?: SetOptions): this;
-  set(keys: Partial<this>, options?: SetOptions): this;
+  set(keys: object, options?: SetOptions): this;
```

Comment:
Not 100% certain, but looks like: https://github.com/sequelize/sequelize/blob/master/lib/model.js#L3249